src/nbl/asset/utils/CGeometryCreator.cpp

```diff
@@ -1838,38 +1945,109 @@ class Icosphere
 
 core::smart_refctd_ptr<ICPUPolygonGeometry> CGeometryCreator::createIcoSphere(float radius, uint32_t subdivision, bool smooth) const
 {
-	Icosphere IcosphereData(radius, subdivision, smooth);
-	
-	return_type icosphereGeometry;
 
-	constexpr size_t vertexSize = sizeof(IcosphereVertex);
+	Icosphere icosphere(radius, subdivision, smooth);
+
+	auto retval = core::make_smart_refctd_ptr<ICPUPolygonGeometry>();
+	retval->setIndexing(IPolygonGeometryBase::TriangleList());
+
+	using namespace hlsl;
+
+	// Create indices
+	{
+		auto indexBuffer = asset::ICPUBuffer::create({ icosphere.getIndexSize() });
+		memcpy(indexBuffer->getPointer(), icosphere.getIndices(), indexBuffer->getSize());
+
+		shapes::AABB<4,Icosphere::index_t> aabb;
+		aabb.minVx[0] = 0;
+		aabb.maxVx[0] = icosphere.getPositionCount() - 1;
+
+		static_assert(sizeof(Icosphere::index_t) == 2 || sizeof(Icosphere::index_t) == 4);
+		const auto isIndex16Bit = sizeof(Icosphere::index_t) == 2;
+
+		retval->setIndexView({
+			.composed = {
+				.encodedDataRange = {.u32=aabb},
```

Comment:
set the AABB via a visitor or after you set the index view, because when format is u16, you're assining to `u32` but you should assign to `u16`